### PR TITLE
Security Module

### PR DESCRIPTION
**Pull request by Emadgnia**
- Bug Fix on Authentication: If a user has some clients that uses this library, they might share a keychain, now you have to inject an identifier for keychain to save a token properly
- Feature added to Networking to be able to support sending Mulitpart requests
- UIComponent for ImagePicker has been added to the UIComponenets module
- Module SecurityHandler has been added for doing the following:
  - CoreDataHelper
  - CryptoHelper
  - KeychainHelper
- Tests and Documentation has been added to SecurityHandler

### CODE REVIEW

**Review by Nicolaidam — APPROVED** [review]
Looks good, could not find anything critical. 
Left a few comments that you can maybe use

File: Sources/Authentication/AuthenticationHandler.swift [hunk 1]
```diff
@@ -63,10 +65,11 @@ public final class AuthenticationHandler: NSObject, ObservableObject {
     /// ````
     /// - Parameter configuration: One way you can using this lib, look at example
     /// - Parameter contextProvider: Provide a window to show the login, look at example
```
[the remark this answers is not in the document]
Comment (Emadgnia) [reply Sources/Authentication/AuthenticationHandler.swift:67]: Done

File: Sources/Authentication/AuthenticationHandler.swift [hunk 2]
```diff
@@ -85,10 +88,11 @@ public final class AuthenticationHandler: NSObject, ObservableObject {
     /// ````
     /// - Parameter automatedLoginModel: One way you can using this lib, look at example
```
[the remark this answers is not in the document]
Comment (Emadgnia) [reply Sources/Authentication/AuthenticationHandler.swift:90]: Done

File: Sources/Networking/MultipartFormData.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+//  File.swift
+//  
+//
+//  Created by Emad Ghorbaninia on 14/04/2023.
+//
+
+import Foundation
+public extension Networking {
+    struct MultipartFormData {
+        public init() {}
```
[the remark this answers is not in the document]
Comment (Emadgnia) [reply Sources/Networking/MultipartFormData.swift:11]: Done

File: Sources/SecurityHandler/CoreDataHelper/CoreDataHelper.swift [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        do {
+            let fetchedData = try viewContext.fetch(fetchRequest)
+            for data in fetchedData {
+                if data.isExpired {
+                    viewContext.delete(data)
+                }
+            }
+            try viewContext.save()
+        } catch {
+            print("Error fetching images: \(error)")
```
[the remark this answers is not in the document]
Comment (Emadgnia) [reply Sources/SecurityHandler/CoreDataHelper/CoreDataHelper.swift:116]: No, it does not matter, if there is an error for fetching means there is nothing to be removed right?

File: Sources/SecurityHandler/CryptoHelper/CryptoHelper.swift [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                                                           keychain: KeychainHelperProtocol.Type) throws -> Data {
+            try getEncryptedKeyByUsingRSAPublicKeyInternal(publicKeyData: publicKeyData, symmetricKeyIdentifier: symmetricKeyIdentifier, keychain: keychain)
+        }
+        private static func getEncryptedKeyByUsingRSAPublicKeyInternal(publicKeyData:Data? = nil,
+                                                                       symmetricKeyIdentifier: String,
+                                                                       keychain: KeychainHelperProtocol.Type) throws -> Data {
+            try getSymmetricKey(symmetricKeyIdentifier: symmetricKeyIdentifier, keychain: keychain)
+            let rsaKey = try CryptoHelper.rsaPublicKeyFromPEM(derKey: publicKeyData)
+            let aesKeyData = try CryptoHelper.loadSymmetricKeyData(symmetricKeyIdentifier: symmetricKeyIdentifier, keychain: keychain)
+            //TODO: use rsaEncryptionOAEPSHA256 in algorithm instead.
```
[the remark this answers is not in the document]
Comment (Emadgnia) [reply Sources/SecurityHandler/CryptoHelper/CryptoHelper.swift:72]: Yes, It's more safe!

File: Sources/SecurityHandler/CryptoHelper/CryptoHelper.swift [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+    private static func generateAndStoreSymmetricKey(symmetricKeyIdentifier: String,
+                                                     keychain: KeychainHelperProtocol.Type) throws -> SymmetricKey {
+        let key = SymmetricKey(size: .bits256)
+        try storeSymmetricKey(key, symmetricKeyIdentifier: symmetricKeyIdentifier, keychain: keychain)
+        return key
+    }
+    
+    private static func loadRSAPublicKey() throws -> Data {
+        let resourceFileName = "key"
```
[the remark this answers is not in the document]
Comment (Emadgnia) [reply Sources/SecurityHandler/CryptoHelper/CryptoHelper.swift:145]: Moved it in test

**Review by Nicolaidam — APPROVED** [review]

**Pull request merged**